x-pack/plugins/security_solution/public/common/components/exceptions/add_exception_comments.tsx

```diff
@@ -0,0 +1,128 @@
+import { useCurrentUser } from '../../lib/kibana';
+import { getFormattedComments } from './helpers';
+
+interface AddExceptionCommentsProps {
+  exceptionItemComments?: Comment[];
+  newCommentValue: string;
+  newCommentOnChange: (value: string) => void;
+}
+
+const commentAccordionButtonClassname = 'exceptionCommentAccordionButton';
+
+const MyAvatar = styled(EuiAvatar)`
+  ${({ theme }) => css`
+    margin-right: ${theme.eui.paddingSizes.m};
+  `}
+`;
+
+const CommentAccordion = styled(EuiAccordion)`
+  ${({ theme }) => css`
+    .${commentAccordionButtonClassname} {
+      color: ${theme.eui.euiColorPrimary};
+      padding: ${theme.eui.paddingSizes.m} 0;
+    }
+  `}
+`;
+
+export const AddExceptionComments = memo(function AddExceptionComments({
+  exceptionItemComments,
+  newCommentValue,
+  newCommentOnChange,
+}: AddExceptionCommentsProps) {
+  const [shouldShowComments, setShouldShowComments] = useState(false);
+  const currentUser = useCurrentUser();
+
+  const handleOnChange = useCallback(
+    (event: React.ChangeEvent<HTMLTextAreaElement>) => {
+      newCommentOnChange(event.target.value);
+    },
+    [newCommentOnChange]
+  );
+
+  const handleTriggerOnClick = useCallback((isOpen: boolean) => {
+    setShouldShowComments(isOpen);
+  }, []);
+
+  const shouldShowAccordion: boolean = useMemo(() => {
+    return (
+      exceptionItemComments !== undefined &&
+      exceptionItemComments !== null &&
+      exceptionItemComments.length > 0
+    );
+  }, [exceptionItemComments]);
+
+  const commentsAccordionTitle = useMemo(() => {
+    if (exceptionItemComments && exceptionItemComments.length > 0) {
+      return (
+        <EuiText size="s" data-test-subj="addExceptionCommentsAccordionButton">
+          {!shouldShowComments
+            ? i18n.COMMENTS_SHOW(exceptionItemComments.length)
+            : i18n.COMMENTS_HIDE(exceptionItemComments.length)}
+        </EuiText>
+      );
+    } else {
+      return null;
+    }
+  }, [shouldShowComments, exceptionItemComments]);
+
+  const formattedComments = useMemo((): EuiCommentProps[] => {
+    if (exceptionItemComments && exceptionItemComments.length > 0) {
+      return getFormattedComments(exceptionItemComments);
+    } else {
+      return [];
+    }
+  }, [exceptionItemComments]);
+
+  return (
+    <div>
+      {shouldShowAccordion && (
+        <CommentAccordion
+          id={'add-exception-comments-accordion'}
+          buttonClassName={commentAccordionButtonClassname}
+          buttonContent={commentsAccordionTitle}
+          data-test-subj="addExceptionCommentsAccordion"
+          onToggle={(isOpen) => handleTriggerOnClick(isOpen)}
+        >
+          <EuiCommentList comments={formattedComments} />
+        </CommentAccordion>
+      )}
+      <EuiFlexGroup gutterSize={'none'}>
+        <EuiFlexItem grow={false}>
+          <MyAvatar
+            name={currentUser !== null ? currentUser.username.toUpperCase() ?? '' : ''} // TODO: Still need to standardize this
```

Comment:
Is there a case where there wouldn't be a username accessible here? If there isn't one does that mean they don't have certain access rights and would we at that point not let them add an exception at all?